pkg/kmutex/kmutex.go

```diff
@@ -119,12 +139,15 @@ func (m *KMutex) Lock(k string) bool {
 }
 
 // Unlock release the lock.
+// If the key does not exist, return immediately.
+// Otherwise, fill up the chan to broadcast that someone released the lock.
 func (m *KMutex) Unlock(k string) {
 	m.Mutex.Lock()
 	defer m.Mutex.Unlock()
 
-	v, ok := m.keys[k]
+	v, ok := m.lockedKeys[k]
 	if ok && len(v.c) == 0 {
+		// filled up the chan to identify caller has released the lock.
 		v.c <- struct{}{}
 	}
 }
```

Comment:
Is it better that free unused lockedKeys in this `Unlock` method than using a extra go routine?
https://github.com/alibaba/pouch/blob/f56461421b3a2148db94d825a5e715fb156e2972/pkg/kmutex/kmutex.go#L41-L52

Reply:
Actually this is a legacy function which is designed by @skoo87 .
Could you help explain a little bit more about the goroutine usage here?